src/Bucket/bucket_parameterizations.jl

```diff
@@ -0,0 +1,93 @@
+"""
+     heaviside(x::FT)::FT where {FT}
+
+Computes the heaviside function.
+"""
+function heaviside(x::FT)::FT where {FT}
+    if x > eps(FT)
+        return FT(1.0)
+    else
+        return FT(0.0)
+    end
+end
+
+"""
+    surface_albedo(α_soil::FT, α_snow::FT, S::FT, S_c::FT)::FT where {FT <: AbstractFloat}
+
+Returns the surface albedo, linearly interpolating between the albedo
+of snow and of soil, based on the snow water equivalent S relative to
+the parameter S_c.
+
+This is taken from Lague et al 2019.
+"""
+function surface_albedo(
+    α_soil::FT,
+    α_snow::FT,
+    S::FT,
+    S_c::FT,
+)::FT where {FT <: AbstractFloat}
+    safe_S::FT = max(S, eps(FT))
+    return (FT(1.0) - S / (S + S_c)) * α_soil + S / (S + S_c) * α_snow
+end
+
+"""
+    infiltration_at_point(W::FT, M::FT, P::FT, E::FT, W_f::FT)::FT where {FT <: AbstractFloat}
+
+Returns the soil infiltration given the current water content of the bucket W, 
+the snow melt volume flux M, the precipitation volume flux P, the liquid evaporative volume
+flux E, and the bucket capacity W_f.
+
+Extra inflow when the bucket is at capacity runs off.
+"""
+function infiltration_at_point(
+    W::FT,
+    M::FT,
+    P::FT,
+    E::FT,
+    W_f::FT,
+)::FT where {FT <: AbstractFloat}
+    if (W > W_f) & (P + M > E) # Equation (4b) of text
```

Comment:
double check this if E is negative - condensation

Reply:
This is fine, slightly redefined it in the overleaf to have infiltration I satisfy \vec{F} = -I z^, so infiltration is related to the boundary flux directly.